openapi/workflow_execution_service.swagger.yaml

```diff
@@ -88,11 +88,27 @@ paths:
           in: query
           required: false
           type: string
-        - name: tag_search
+        - name: tag_key
           description: |-
             OPTIONAL
-            For each key, if the key's value is empty string then match workflows that are tagged with
-            this key regardless of value.
+            Filter returned workflow runs to only return those where the specified key is present in "tags".
+          in: query
+          required: false
+          type: string
+        - name: tag_value
```

Comment:
Does OpenAPI specify how `type: object` is encoded when it is part of a query?  The main reason I proposed doing it this way is to avoid jamming ugly url-encoded json in the query portion

Reply:
The default style (`form`) and "explode" behavior (`true`) for object [parameter serialization](https://swagger.io/docs/specification/serialization/) in queries would apparently look like this:

`Object id = {"role": "admin", "firstName": "Alex"}` => `/users?role=admin&firstName=Alex`

You can specify more fine-grained and/or complex types for the parameter with schema (example from [this page](https://swagger.io/docs/specification/describing-parameters/):

```yaml
parameters:
  - in: query
    name: filter
	
    # Wrap 'schema' into 'content.<media-type>'
    content:
      application/json:  # <---- media type indicates how to serialize / deserialize the parameter content
        schema:
          type: object
          properties:
            type:
              type: string
            color:
              type: string
```

... but I think that only works if we define specific keys for the filter. (?)

Not sure if that answers your question, or avoids encoding issues. I can play around with some examples after today's call to see how it looks.

Reply:
Oh, I see.  I think you're proposing that any query parameter that isn't one of `page_size`, `page_token` or `state` would be considered a filter on tags.  That seems quite reasonable.

Reply:
An alternative to URL-encoding a JSON object and including it as a query parameter is to put the object in the request body (and use POST instead of GET for the HTTP verb).  The advantage is that if the object becomes large the URL will not run up against the maximum URL length.